schema/index.graphql

```diff
@@ -171,16 +281,19 @@ A warehouse
 type Warehouse {
   id: ID!
   name: String!
-  openingTimes: [OpeningTime!]!
+  openingTimes: [DateTime!]!
+  closingTimes: [DateTime!]
 }
 
 """
-Opening times of a warehouse/store
+Opening/closing times of a warehouse/store
+When no explicit date is specified dayOfWeek needs to be specified
 """
-type OpeningTime {
-  dayOfWeek: String!
+type DateTime {
+  dayOfWeek: Int
```

Comment:
I suggest adding a graphql directive to make explicit that either `dayOfWeek` or `date` needs to be specified.
Use case for day of week is for recurring times (e.g.: every monday from 07:00 - 18:00), while date serves for exceptional cases (e.g.: except on 24.12 from 14:00 - 24:00)